### PR TITLE
[CI] Refactor external tests into its own workflow.

### PR DESCRIPTION
**Pull request by christianparpart**

### CODE REVIEW

File: .github/workflows/codechecker-analysis.yml [hunk 1]
```diff
@@ -28,6 +28,11 @@ jobs:
     steps:
       - name: "Check out repository"
         uses: actions/checkout@v2
+      - name: ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          key: ccache-ubuntu2004-external-tests
+          max-size: 256M
```
Comment (whisperity) [inline .github/workflows/codechecker-analysis.yml:35]: Please don't. `CodeChecker log` needs the compiler invocations... I *think* it *can* synergise with CCache, but there are usually some issues cropping up here and there, I'd prefer if it did not intervene.

File: scripts/ci-prepare-contour.sh [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
+CMAKE_BUILD_TYPE="${CMAKE_BUILD_TYPE:-RelWithDebInfo}"
 
 prepare_build_ubuntu()
 {
-   mkdir -pv ${BUILD_DIR}
-   cd ${BUILD_DIR}
-   cmake .. \
+   cmake \
       -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
-      -DCMAKE_CXX_COMPILER="${CXX}" \
```
Comment (whisperity) [inline scripts/ci-prepare-contour.sh:11]: Where did this line go, and why did it disappear?

**Review by whisperity — CHANGES_REQUESTED** [review]

**Comment by whisperity** [comment]
Also, `notcurses-install-deps.sh` should be moved into some better place than the `workflows` directory... Maybe into the top-level `scripts`, or just one directory out, but don't store other files than the workflows and such in the directory.

More of a matter of good appearance.

**Review by uspasojevic96 — DISMISSED** [review]

**Comment by whisperity** [comment]
You also most likely **aren't** using CCache:

Screenshot_20211124_115820

Usually, if CCache is used, the compiler would be from `/usr/lib/ccache/g++-8` or something, usually by prepending `/usr/lib/ccache` in `PATH`.

Unless that custom action does some sort of magic...

In general, you can validate the use of CCache (and its good practice) to add `ccache -s` ("print stats") before and after the build, either as its own action, or just as an additional command in the list of commands that constitute the "build" action.

File: .github/workflows/codechecker-analysis.yml (hunk 1 above)
Comment (christianparpart) [reply .github/workflows/codechecker-analysis.yml:35]: aaah! That is not even meant to be there. haha. Will remove that!

File: scripts/ci-prepare-contour.sh (hunk 2 above)
Comment (christianparpart) [reply scripts/ci-prepare-contour.sh:11]: I removed it in order to get cmake happy finding the compiler. TBH, I'm not sure we need or do not need this line.

**Comment by christianparpart** [comment]
> Usually, if CCache is used, the compiler would be from `/usr/lib/ccache/g++-8` or something, usually by prepending `/usr/lib/ccache` in `PATH`.
> 
> Unless that custom action does some sort of magic...
> 
> In general, you can validate the use of CCache (and its good practice) to add `ccache -s` ("print stats") before and after the build, either as its own action, or just as an additional command in the list of commands that constitute the "build" action.

I think this is because ccache is looked up later, but will definitely be used for building. It should have been used earlier already. That's definitely true. the `ccache -s` is actually performed as post job, just inspect one of those build jobs and you'll find it at the bottom.

**Comment by whisperity** [comment]
> the `ccache -s` is actually performed as post job, just inspect one of those build jobs and you'll find it at the bottom.

Yeaaah let's say the interface is not the most intuitive...

**Review by whisperity — APPROVED** [review]

**Pull request merged**